### PR TITLE
[restapi] fixes #86 Add Python client for skycoin

### LINKED ISSUE #86: Python API wrapper for the Skycoin node API
## Expected Behavior

Access to the Skycoin node REST API implemented in Python code .

## Actual Behavior

Right now this is only possible using code generated by SWIG , which requires memory handles

## Specifications

  - Version: 0.25.0
  - Platform: all
  - Subsystem: all exported API sets

## Possible Solution

~Feel free to use e.g. [requests](https://pypi.org/project/requests/)~

- [x] Use Skycoin node Swagger spec to generate initial Python REST API client
- [x] Add `make` target to automate generation in `./build`
- [x] Aforementioned target should copy / merge relevant files into source code tree
- [x] README explaining the process
- [x] Mention in CHANGELOG
- [x] Patch the result Swagger-gen REST client to support specifying node address
- [x] Update package metadata
- [x] Run basic Swagger tests in Travis
- [x] Unify test code to use either `py.test` (preferred) or `nosetests`
- [x] Beware of CSRF auth

Given the initial commits ...

- [x] Besides rebase initial patch against `0.25.dev` branch

### PR DESCRIPTION
**Pull request by stdevMac**
Changes: #86 
- Working to add skycoin client from a swagger specification

Does this change need to mentioned in CHANGELOG.md?

Yes

### CODE REVIEW

**Comment by olemis** [comment]
@stdevMac pls sign the CLA . Needed for merge .

**Comment by olemis** [comment]
I'll add in #86 some tasks to do .

**Comment by stdevMac** [comment]
@olemis I already registered to the CLA

File: .travis/install-linux.sh [hunk 1]
```diff
@@ -18,6 +18,8 @@ chmod +x "$HOME/bin/gimme"
 # Install Python libraries
 $PIP install --upgrade pip setuptools tox-travis
 $PIP install -r "$REPO_ROOT/requirements.dev.txt"
+$PIP install -r "$REPO_ROOT/pysky_api/requirements.txt"
```
Comment (olemis) [inline .travis/install-linux.sh]: skyapi

File: Makefile [hunk 2]
```diff
@@ -36,6 +38,11 @@ configure: ## Configure build environment
 	mkdir -p $(BUILDLIBC_DIR) $(BIN_DIR) $(INCLUDE_DIR)
 	mkdir -p $(DIST_DIR)
 
+generate-python-Client: ## Configure build environment
```
Comment (olemis) [inline Makefile]: generate-client

Comment (olemis) [reply Makefile]: what does it do exactly ?

File: Makefile [hunk 3]
```diff
@@ -66,23 +73,29 @@ build-swig: ## Generate Python C module from SWIG interfaces
 
 develop: ## Install PySkycoin for development
 	$(PYTHON_BIN) setup.py develop
```
Comment (olemis) [inline Makefile]: `PYTHON_BIN` should be just `PYTHON`

File: docs/DefaultApi.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+### Example
+```python
+from __future__ import print_function
+import time
+import swagger_client
+from swagger_client.rest import ApiException
+from pprint import pprint
+
+# Configure API key authorization: csrfAuth
+configuration = swagger_client.Configuration()
```
Comment (olemis) [inline docs/DefaultApi.md]: `swagger_client` ?

File: pysky_api/README.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
+# openapi-client
```
Comment (olemis) [inline pysky_api/README.md]: `openapi-client` ?

File: skycoin/restapi/__init__.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# coding: utf-8
```
Comment (olemis) [inline skycoin/restapi/__init__.py]: Remove `skycoin/restapi` .

Comment (stdevMac) [reply skycoin/restapi/__init__.py]: already removed in 6bad8d161201dfecbc7a6f7b7ccdd15bdc156474

File: pysky_api/README.md (hunk 5 above)
Comment (stdevMac) [reply pysky_api/README.md]: Autogenerated by swagger

File: docs/DefaultApi.md (hunk 4 above)
Comment (stdevMac) [reply docs/DefaultApi.md]: Autogenerated by swagger

File: Makefile (hunk 3 above)
Comment (stdevMac) [reply Makefile]: Already done in last commit

File: .travis/install-linux.sh (hunk 1 above)
Comment (stdevMac) [reply .travis/install-linux.sh]: Already done in last commit

**Pull request merged**